### PR TITLE
Config updates

### PR DESCRIPTION
**Pull request by dylanlott**
Config 🅱️  handling 📦  updates 👌

The admin node now requires no extra flags to boot up correctly. ⭐️ 

You can start the entire admin node with `go run cmd/admin/main.go` (assuming you've installed everything necessary) 🚀 

# Major updates:
- Looks for an existing config file in $HOME/.storj 🏠 
- Generates a config file from flag defaults if none is found 🎏 
- References config file for settings, sets up viper correctly 🐍 
- Flag precedence is respected in services 🥇 
- Services can call `v := viper.GetViper()` to access all configuration settings, or they can refer to flags instead if they wish. 🙏 

# Minor updates:
- Set sane default for pkg/overlay boltDB default path 🔩 
- Updated pkg/overlay `srvPort` from 8080 to 8082 to avoid a port collision ⛵️ 
- Renamed pkg/pointerdb `pointerdbDB` flag to `pointerdb` for clarity in the config file and flag usage ☝️

---
This change is [Reviewable](https://reviewable.io/reviews/storj/storj/141)

### CODE REVIEW

File: pkg/overlay/service.go [hunk 1]
```diff
@@ -28,13 +31,16 @@ var (
 	options                                                                                  peertls.TLSFileOptions
 )
 
+var home, _ = homedir.Dir()
+var boltDBPath = filepath.Join(home, ".storj", "overlaydb.db")
```
Comment (jtolds) [inline pkg/overlay/service.go]: these should probably be in an init method. i don't know if the go language spec guarantees that these two variables are defined in order

Comment (jtolds) [reply pkg/overlay/service.go]: like

```golang
var (
  home, boltDBPath string
)

func init() {
  home, _ = homedir.Dir()
  boltDBPath = filepath.Join(...)
}
```

perhaps even better

```
func defaultBoltDBPath() string {
   ...
}
```
no global variables, and just use the return value in the call to define the flag

Comment (dylanlott) [reply pkg/overlay/service.go]: Done.

File: pkg/overlay/service.go [hunk 2]
```diff
@@ -139,7 +147,7 @@ func (s *Service) Process(ctx context.Context, _ *cobra.Command, _ []string) (
 		return err
 	}
 
-	kad, err := kademlia.NewKademlia(id, []proto.Node{*in}, "0.0.0.0", localPort)
+	kad, err := kademlia.NewKademlia(id, []proto.Node{*in}, "0.0.0.0", viper.GetString("localport"))
```
Comment (jtolds) [inline pkg/overlay/service.go]: every config value inside pkg/overlay should probably start with "overlay." perhaps? like "overlay.localport"?

File: pkg/overlay/service.go [hunk 3]
```diff
@@ -157,26 +165,24 @@ func (s *Service) Process(ctx context.Context, _ *cobra.Command, _ []string) (
 
 	// bootstrap cache
 	var cache *Cache
-	if redisAddress != "" {
-		cache, err = NewRedisOverlayCache(redisAddress, redisPassword, db, kad)
+	if viper.GetString("redisaddress") != "" {
```
Comment (jtolds) [inline pkg/overlay/service.go]: same. what if the pointerdb has a redisaddress?

File: pkg/pointerdb/service.go [hunk 4]
```diff
@@ -21,7 +21,7 @@ import (
 
 var (
 	port   = flag.Int("port", 8080, "port")
-	dbPath = flag.String("pointerdbDB", "pointerdb.db", "pointerdb db path")
+	dbPath = flag.String("pointerdb", "pointerdb.db", "pointerdb db path")
```
Comment (jtolds) [inline pkg/pointerdb/service.go]: lol are you sure you don't like pointerdbDB?

File: pkg/overlay/service.go (hunk 3 above)
Comment (dylanlott) [reply pkg/overlay/service.go]: I would argue we should never have more than one instance of redis running, across the entire app, so they should both use the same redis instance.

File: pkg/pointerdb/service.go (hunk 4 above)
Comment (dylanlott) [reply pkg/pointerdb/service.go]: I made this change with judicious prejudice.

**Pull request merged**